lib/galaxy/jobs/runners/util/condor/__init__.py

```diff
@@ -67,15 +68,21 @@ def build_submit_description(executable, output, error, user_log, query_params):
     return "\n".join(submit_description)
 
 
-def condor_submit(submit_file):
+def condor_submit(submit_file, prefix: Optional[str] = None, command: Optional[str] = None):
```

Comment:
Is there somewhere a case when you need to set prefix and command to something else than `self.runner_params.get("condor_submit_cmd")` ? If not can you not remove the optional arguments?

Or if you want to keep them, wouldn't it make more sense to do something like `prefix = prefix or self.runner_params.get("prefix")`

Reply:
> prefix = prefix or self.runner_params.get("prefix")

I cannot do this because `condor_submit` is not a _bound method_ and thus there is no `self` in its scope.

> Is there somewhere a case when you need to set prefix and command to something else than `self.runner_params.get("condor_submit_cmd")` ? If not can you not remove the optional arguments?

No, but this is not an arbitrary decision nevertheless. If you really want to enter in this discussion we can, but I really do not think it is worth the debate. I think we should rather be keeping our heads busy with, for example, how is the monitoring supposed to be going to work during the migration.